### ISSUE TITLE
feat[dev]: ✨ implement A11y improvements for Tag Template

### ISSUE THREAD

**Issue by yoezlem** [solid-design-system/solid#1525]
## Description
This task describes the Storybook implementation for the Tag template as part of the A11y improvements outlined in Epic https://github.com/solid-design-system/solid/issues/1465.

A11y improvements for the Tag template can be found here: [Tag Notes](https://www.notion.so/matuzo/Tag-10ef3f6e1bc080c58ea8d069aa224315).

-----------------

## Comment
**Problem:** The button with the stars has no accessible name.
**Solution:** Add a label (use sr-only class)

## DoR
- [x] Item has been estimated by the team
- [x] Item dependencies have been identified and documented

## DoD
- [ ] Documentation has been created/updated (if applicable)
- [ ] Migration Guide has been created/updated (if applicable)
- [ ] Relevant E2E tests (Features, A11y, Bug fixes) are created/updated
- [ ] Relevant stories (Features, A11y) are created/updated
- [ ] Implementation works successfully on `feature` branch
